TsJournalTrello.gs

```diff
@@ -0,0 +1,361 @@
+      return
+    }    
+    Log_.info('Trello Card Id: ' + trelloBoardId)
+        
+  } //if (trelloBoardUrl === '')
+      
+  //Open the Journal, loop through the paragraphs until the Heading 1 is found
+  //The next paragraph is the Trello Card Name, check it hasnt already been processed.
+  //If it hasnt, change the style to Heading 2 and add a Hyperlink to the Trello Card 
+  var journal = DocumentApp.openById(DocId)
+  var DocUrl = journal.getUrl()
+  var npBookmark = null
+ 
+  if (journal === null) { 
+  
+    Log_.warning('Invalid Journal ID' + DocId)
+   
+    return
+  } else {
+   
+    //Process the journal, find the next tello card task, hyperlink the heading
+    var [trelloCardTitle, npBookmarkId] = processJournal_(journal, trelloBoardId)
+          
+    if (trelloCardTitle == null) {
+      Log_.warning('Either no trello card title found or trello card found on trello board')
+      return
+    } else if (npBookmarkId == null) {
+      Log_.warning('No Bookmark Found')    
+      return
+    } else {
+      //Add the link to the timesheet
+      processTimesheet_(timesheetUrl, npBookmarkId, DocUrl, trelloCardTitle)    
+    }
+  }
+                
+  return SUCCESS  
+}// linkHeaders() 
+
+
+function processJournal_(journal, trelloBoardId) {
+          
+  var paragraphs = journal.getBody().getParagraphs()
+  var ProcessJournalReturn = null
+  var Heading1Paragraphs = []
+  
+  for(var i in paragraphs) { 
+    if (paragraphs[i].getHeading() === DocumentApp.ParagraphHeading.HEADING1) {
+      Heading1Paragraphs.push(i)
+    }
+  }
+  
+  var lastHeading1Paragraph = Heading1Paragraphs[Heading1Paragraphs.length -1]
+  var trelloCardParagraph = paragraphs[Number(lastHeading1Paragraph) + 1]
+  var trelloCardTitle = trelloCardParagraph.getText()
+  
+  if (trelloCardParagraph.getHeading() !== DocumentApp.ParagraphHeading.HEADING2) {
+    var trelloCardUrl = getTrelloCardUrl_(trelloBoardId, trelloCardTitle)
+    
+      if (trelloCardUrl == null) {
+  
+        Log_.warning('Card Name ' + trelloCardTitle + ' not found in list of Trello Boards')
+        return [null, null]
+      } else {
+  
+        var sectionPos = journal.newPosition(trelloCardParagraph, 0);
+        var npBookmark = journal.addBookmark(sectionPos)
+        
+        // Set the next paragraph to heading 2 and add the Trello Card Title URL link
+        trelloCardParagraph.setHeading(DocumentApp.ParagraphHeading.HEADING2)
+        trelloCardParagraph.setLinkUrl(trelloCardUrl)
+      
+        ProcessJournalReturn = [trelloCardTitle, npBookmark.getId()]
+        return ProcessJournalReturn 
+     }
+  } else {
+    Log_.warning('No new trello card found')
+    return [null, null]
+  }
+       
+}
+          
+function processTimesheet_(timesheetUrl, npBookmarkId, DocUrl, trelloCardTitle) {
+  
+  if (timesheetUrl === '') {
+
+    Log_.warning('No Timesheet URL found in Org Spreadsheet')
+    return
+    
+  } else {
```

Comment:
The else is redundant as it'll only get past this block if the condition is true, so it'll avoid some indentation and you can put all the arg checks at the beginning. No need to mix "early returns" and loads of if/elses. All the meat of the code won't then need to be indented.